beacon-chain/blockchain/process_block.go

```diff
@@ -405,6 +403,25 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.SignedBeaconBlo
 	return nil
 }
 
+func (s *Service) tryPublishLightClientFinalityUpdate(ctx context.Context, signed interfaces.SignedBeaconBlock, finalized *forkchoicetypes.Checkpoint, postState state.BeaconState) {
+	if finalized.Epoch > s.lastPublishedLightClientEpoch {
+		config := params.BeaconConfig()
+		if finalized.Epoch > config.AltairForkEpoch {
+			syncAggregate, err := signed.Block().Body().SyncAggregate()
+			if err == nil {
+				if syncAggregate.SyncCommitteeBits.Count()*3 >= config.SyncCommitteeSize*2 {
+					_, err := s.sendLightClientFinalityUpdate(ctx, signed, postState)
+					if err != nil {
+						log.WithError(err)
```

Comment:
should it be more fatal than this? I guess in general we took the stance that light client errors just get logged? Is that fair? May make sense to check how eth1 is handling it

Reply:
This is part of `stream publishing` flow.   
So `not publishing` is the hardest penalty.    We want to keep the stream (serving other purposes as well).   So `error out` isn't an option here.